spalloc/_keepalive_process.py

```diff
@@ -0,0 +1,65 @@
+from spalloc.protocol_client import ProtocolClient, ProtocolTimeoutError
+import sys
+import threading
+
+
+def _wait_for_exit(keepalive):
+    for line in sys.stdin:
+        if line.strip() == "exit":
+            keepalive.stop()
+
+
+class KeepAliveProcess(object):
+
+    def __init__(self):
+        self._stop = threading.Event()
+        self._running = threading.Event()
+
+    def stop(self):
+        self._stop.set()
+
+    def wait_for_start(self):
+        self._running.wait()
+
+    def run(self, hostname, port, job_id, keepalive, timeout, reconnect_delay):
+        """Background keep-alive thread."""
```

Comment:
This is a deceptive comment. This is actually the _foreground_ thread of this process; the background thread is the one that's listening to stdin. That's making this code a little confusing.

Reply:
In fact, thinking about it the whole thing shouldn't really be a class here as it's not getting any benefit from the structure at all.